beacon_node/beacon_chain/src/schema_change/migration_schema_v24.rs

```diff
@@ -0,0 +1,292 @@
+                        // We choose to not compute states during the epoch with block replay for
+                        // simplicity. Therefore we can only support a hot heriarchy config where the
+                        // lowest layer is >= 5.
+                        if slot % T::EthSpec::slots_per_epoch() != 0 {
+                            return Err(Error::MigrationError(
+                                "Hot hierarchy config lowest value must be >= 5".to_owned(),
+                            ));
+                        }
+
+                        // Load the full state and re-store it as a snapshot or diff.
+                        let state = get_full_state_v22(&db.hot_db, &state_root, &db.spec)?
+                            .ok_or(Error::MissingState(state_root))?;
+
+                        // Store immediately so that future diffs can load and diff from it.
+                        let mut ops = vec![];
+                        // We must commit the hot state summary immediatelly, otherwise we can't diff
+                        // against it and future writes will fail. That's why we write the new hot
+                        // summaries in a different column to have both new and old data present at
+                        // once. Otherwise if the process crashes during the migration the database will
+                        // be broken.
+                        db.store_hot_state_summary(&state_root, &state, &mut ops)?;
+                        db.store_hot_state_diffs(&state_root, &state, &mut ops)?;
+                        db.hot_db.do_atomically(ops)?;
+                        diffs_written += 1;
+                    }
+                    StorageStrategy::ReplayFrom(_) => {
+                        // Optimization: instead of having to load the state of each summary we load x32
+                        // less states by manually computing the HotStateSummary roots using the
+                        // computed state dag.
+                        //
+                        // No need to store diffs for states that will be reconstructed by replaying
+                        // blocks.
+                        //
+                        // 2. Convert the summary to the new format.
+                        let previous_state_root = if state_root == split.state_root {
+                            Hash256::ZERO
+                        } else {
+                            state_summaries_dag
+                                .previous_state_root(state_root)
+                                .map_err(|e| {
+                                    Error::MigrationError(format!(
+                                        "error computing previous_state_root {e:?}"
+                                    ))
+                                })?
+                        };
+
+                        let diff_base_state_root =
+                            if let Some(diff_base_slot) = storage_strategy.diff_base_slot() {
```

Comment:
Don't need to call this function because we can read the slot from the `ReplayFrom` which is matched above

Reply:
Fixed in https://github.com/sigp/lighthouse/pull/6750/commits/088cfa03deec97ca56eb6bea4adaae35a2fa37a6